### DISCUSSION TITLE
Latest update has safety checker enabled!

### DISCUSSION THREAD

**Discussion by levicki** [d8ahazard/sd_dreambooth_extension#1350] [Q&A]
@d8ahazard 

Today I updated to the latest version of the extension and I tried training a LoRA on some NSFW content.

The class images generated were all black, and most of them weren't even being saved to the classifiers output folder.

I checked, and under the `model\dreambooth\MODEL_NAME\working` there is a `safety_checker` folder with 1.13 GB `model.safetensors` file and `config.json` with commit hash `cb41f3a270d63d454d385fc2e4f571c487c253c5` and name/path `CompVis/stable-diffusion-safety-checker`.

Also, I noticed that `model_index.json` in the `working` folder now contains:
```
{
  "_class_name": "StableDiffusionPipeline",
  "_diffusers_version": "0.19.3",
  "feature_extractor": [
    "transformers",
    "CLIPFeatureExtractor"
  ],
  "requires_safety_checker": true,
  "safety_checker": [
    "stable_diffusion",
    "StableDiffusionSafetyChecker"
  ],
# cut for brevity
```
Before it looked like this:
```
    "_class_name": "StableDiffusionPipeline",
    "_diffusers_version": "0.16.1",
    "feature_extractor": [
        null,
        null
    ],
    "requires_safety_checker": null,
    "safety_checker": [
        null,
        null
    ],
# cut for brevity
```
What in the name of `$DEITY` is going on here?!?

**Comment by levicki** [comment]
A workaround is to manually edit the `model_index.json` and change `requires_safety_checker` to `false`, but having to do that every time we want to create new NSFW LoRA would really be annoying. Hopefully it can be fixed quickly.

**Reply by Prometheus3D** [reply]
Setting requires_safety_checker to false itself didn't seem to work.
I was able to get it fixed with this:

```
  "requires_safety_checker": null,
  "safety_checker": [
	null,
	null
  ],
```

**Reply by shaguru** [reply]
Thanks! this did it for me!

**Comment by morphinapg** [comment]
Interesting. Does this only affect class images? My training does not appear to have been impacted, or at least not much if so. I will try with the above workaround to see if it improves my future models though. I don't use class images and I've been training full models instead of Lora, so idk if those affect anything.

Seems like it would be easy to implement a fix with an checkbox to opt out of the safety checker, but there are quite a few issues with this extension that the bot keeps marking as stale because nothing seems to be being done to fix them. Is this extension still actively being worked on?

**Comment by levicki** [comment]
@morphinapg It affects class image generation. Training is affected if you use class images, because it will train against fully black class images instead of reference images of subjects generated from `[filewords]`.

**Reply by morphinapg** [reply]
Is the safety checker included when the checkpoint is compiled though?

**Reply by levicki** [reply]
It is not included in the checkpoint, but its files are copied to the model working folder (wastes 1.13 GB of space for each training model you create).

**Reply by morphinapg** [reply]
Looks like training is working after deleting those files and changing the above lines. 

Would love to know if there was a temporary change I could make in the code to not need to do that every time though, until the repo gets updated.

**Comment by levicki** [comment]
Relevant issue here: https://github.com/d8ahazard/sd_dreambooth_extension/issues/1352
